test/unittest/tests/block/test_block_addr.cpp

```diff
@@ -0,0 +1,174 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+// This file unit tests block manager functions relating to packing and unpacking address cookies.
+
+#include "wt_internal.h"
+#include <catch2/catch.hpp>
+#include <iostream>
+#include <vector>
+
+void
+unpack_addr_cookie_and_check(const uint8_t *packed, uint32_t block_allocsize, wt_off_t pack_offset,
+  uint32_t pack_size, uint32_t pack_checksum)
+{
+    uint64_t o = 0, s = 0, c = 0;
+    REQUIRE(__wt_vunpack_uint(&packed, 0, &o) == 0);
+    REQUIRE(__wt_vunpack_uint(&packed, 0, &s) == 0);
+    REQUIRE(__wt_vunpack_uint(&packed, 0, &c) == 0);
+
+    // Adjust the unpacked values as the block manager also does this to avoid storing large
+    // offsets.
+    wt_off_t unpacked_offset = (s == 0) ? (wt_off_t)o : (wt_off_t)(o + 1) * block_allocsize;
```

Comment:
This will be annoying to implement but these should all be C++ style static casts.